packages/core/src/awsService/sagemaker/sshConfig.ts

```diff
@@ -0,0 +1,366 @@
+        if (configState.hasSshSection && configState.isOutdated) {
+            // Section is outdated ask user to update it before validation
+            const shouldUpdate = await this.promptToUpdateSshConfig()
+
+            if (shouldUpdate) {
+                try {
+                    // Remove the outdated section
+                    await this.removeSshConfigSection()
+                    // Write the new section
+                    await this.writeSectionToConfig(proxyCommand)
+                    getLogger().info('SSH config: Successfully updated sm_* section')
+                    // Update state to reflect that section now exists and is up to date
+                    configState.hasSshSection = true
+                    configState.isOutdated = false
+                } catch (e) {
+                    // Failed to update, prompt user to fix manually
+                    try {
+                        await this.promptToFixUpdateFailure(e instanceof Error ? e : undefined)
+                        const configOpenedError = new ToolkitError(
+                            `SSH configuration file opened for editing. Fix the issue and try connecting again.`,
+                            {
+                                code: 'SshConfigOpenedForEdit',
+                                details: { configPath: getSshConfigPath() },
+                            }
+                        )
+                        return Result.err(configOpenedError)
+                    } catch (promptError) {
+                        // User cancelled opening the file
+                        if (promptError instanceof CancellationError) {
+                            return Result.err(
+                                ToolkitError.chain(
+                                    e,
+                                    `Failed to update SSH config section. Fix your ~/.ssh/config file manually or remove the outdated ${this.configHostName} section.`,
+                                    {
+                                        code: 'SshConfigUpdateFailed',
+                                        details: {
+                                            configHostName: this.configHostName,
+                                            configPath: getSshConfigPath(),
+                                        },
+                                    }
+                                )
+                            )
+                        }
+                        return Result.err(
+                            ToolkitError.chain(
+                                promptError,
+                                'Unexpected error while handling SSH config update failure',
+                                {
+                                    code: 'SshConfigErrorHandlingFailed',
+                                }
+                            )
+                        )
+                    }
+                }
+            } else {
+                // User declined
```

Comment:
Nit : So many prompts here, worth calling out which one this cancellation maps to.